[BUG]. Flaky test LtrQueryTests.testOnRewrittenQueries

### What is the bug?
com.o19s.es.ltr.query.LtrQueryTests.testOnRewrittenQueries are failing falkily

```
LtrQueryTests > testOnRewrittenQueries FAILED
    java.lang.AssertionError: Scores match with similarity class org.apache.lucene.search.similarities.IBSimilarity expected:<0.48958945> but was:<0.48959094>
        at __randomizedtesting.SeedInfo.seed([EB199384803DDA8B:99B641[86](https://github.com/opensearch-project/opensearch-learning-to-rank-base/actions/runs/14035589736/job/39397286034?pr=151#step:6:87)C260D355]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:577)
        at com.o19s.es.ltr.query.LtrQueryTests.assertScoresMatch(LtrQueryTests.java:362)
        at com.o19s.es.ltr.query.LtrQueryTests.checkModelWithFeatures(LtrQueryTests.java:350)
        at com.o19s.es.ltr.query.LtrQueryTests.testOnRewrittenQueries(LtrQueryTests.java:449)
```

Example failing CI Workflow: [link](https://github.com/opensearch-project/opensearch-learning-to-rank-base/actions/runs/14035589736/job/39397284914?pr=151)

The test passed on Retry for the same workflow.

### How can one reproduce the bug?
Could not reproduce on local machine

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

--- a/src/test/java/com/o19s/es/ltr/query/LtrQueryTests.java
+++ b/src/test/java/com/o19s/es/ltr/query/LtrQueryTests.java
@@ -105,7 +105,7 @@ import ciir.umass.edu.utilities.MyThreadPool;
 @LuceneTestCase.SuppressSysoutChecks(bugUrl = "RankURL does this when training models... ")
 public class LtrQueryTests extends LuceneTestCase {
     // Number of ULPs allowed when checking scores equality
-    private static final int SCORE_NB_ULP_PREC = 1;
+    private static final int SCORE_NB_ULP_PREC = 30000;
 
     private int[] range(int start, int stop) {
         int[] result = new int[stop - start];

